File: LLM-Translations/GPT-4/codenet/Java/Go/s973587235.go
package main

import (
	"fmt"
)

func main() {
	var n, h int
	fmt.Scan(&n)

	minN := 0
	fmt.Scan(&minN)
	minN--

	for i := 0; i < n-1; i++ {
		fmt.Scan(&h)
		if minN == h {
			minN = h
		} else if minN == h-1 {
			minN = h - 1
		} else if minN < h-1 {
			minN = h - 1
		} else if minN > h {
			fmt.Println("No")
			return
		}
	}
	fmt.Println("Yes")
}

// 

File: LLM-Translations/GPT-4/codenet/Java/Python/s973587235.py
import sys

def main():
    n = int(input().strip())
    heights = list(map(int, input().strip().split()))
    minN = heights[0] - 1
    for h in heights[1:]:
        if minN == h:
            minN = h
        elif minN == h - 1:
            minN = h - 1
        elif minN < h - 1:
            minN = h - 1
        elif minN > h:
            print("No")
            return
    print("Yes")

if __name__ == "__main__":
    main()


